<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1009</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Perspective Distortion Remove</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_7">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="maximumSize">
       <size>
        <width>321</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string/>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="infoGroupBox">
         <property name="title">
          <string>Info</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="mousePosLabel">
            <property name="text">
             <string>Mouse</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="mousePosXLineEdit">
            <property name="text">
             <string>0</string>
            </property>
            <property name="frame">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLineEdit" name="mousePosYLineEdit">
            <property name="text">
             <string>0</string>
            </property>
            <property name="frame">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="imageSizeLabel">
            <property name="text">
             <string>Image Size</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="imageSizeXLineEdit">
            <property name="text">
             <string>0</string>
            </property>
            <property name="frame">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLineEdit" name="imageSizeYLineEdit">
            <property name="text">
             <string>0</string>
            </property>
            <property name="frame">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="linesGroupBox">
         <property name="title">
          <string>Orthogonal Lines</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QRadioButton" name="line0RadioButton">
              <property name="text">
               <string>Line 0</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line0x0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line0y0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line0x1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line0y1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="line1RadioButton">
              <property name="text">
               <string>Line 1</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line1x0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line1y0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line1x1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line1y1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QRadioButton" name="line2RadioButton">
              <property name="text">
               <string>Line 2</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line2x0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line2y0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line2x1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line2y1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QRadioButton" name="line3RadioButton">
              <property name="text">
               <string>Line 3</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line3x0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line3y0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line3x1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line3y1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QRadioButton" name="line4RadioButton">
              <property name="text">
               <string>Line 4</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line4x0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line4y0SpinBox">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line4x1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="line4y1SpinBox_2">
              <property name="frame">
               <bool>false</bool>
              </property>
              <property name="decimals">
               <number>0</number>
              </property>
              <property name="maximum">
               <double>99999.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="calculatePushButton">
         <property name="text">
          <string>Calculate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="imageGroupBox">
         <property name="title">
          <string>Image</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QRadioButton" name="inputRadioButton">
            <property name="text">
             <string>Input</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="outputRadioButton">
            <property name="text">
             <string>Output</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="GLImageWidget" name="glImageWidget"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1009</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionZoom_In"/>
    <addaction name="actionZoom_Out"/>
    <addaction name="separator"/>
    <addaction name="actionPoint_Selection"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuRun">
    <property name="title">
     <string>Run</string>
    </property>
    <addaction name="actionRun"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuRun"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Q</string>
   </property>
  </action>
  <action name="actionZoom_In">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="actionZoom_Out">
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="actionPoint_Selection">
   <property name="text">
    <string>Point Selection</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLImageWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>GLImageWidget.h</header>
   <slots>
    <signal>mouseMove(int,int)</signal>
    <signal>newPoint(int,int,int,int)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutDialogShow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>glImageWidget</sender>
   <signal>mouseMove(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onGLMouseMove(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>660</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>glImageWidget</sender>
   <signal>newPoint(int,int,int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onNewPoint(int,int,int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>660</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>calculatePushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onCalculateButtonPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>470</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onInputImageToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>529</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>outputRadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onOutputImageToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>529</y>
    </hint>
    <hint type="destinationlabel">
     <x>506</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line0RadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLine0RadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line1RadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLine1RadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>272</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line2RadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLine2RadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line3RadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLine3RadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>line4RadioButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLine4RadioButtonToggled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>55</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>504</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileOpen()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>aboutDialogShow()</slot>
  <slot>onGLMouseMove(int,int)</slot>
  <slot>onNewPoint(int,int,int,int)</slot>
  <slot>onCalculateButtonPress()</slot>
  <slot>onInputImageToggled(bool)</slot>
  <slot>onOutputImageToggled(bool)</slot>
  <slot>onLine0RadioButtonToggled(bool)</slot>
  <slot>onLine1RadioButtonToggled(bool)</slot>
  <slot>onLine2RadioButtonToggled(bool)</slot>
  <slot>onLine3RadioButtonToggled(bool)</slot>
  <slot>onLine4RadioButtonToggled(bool)</slot>
 </slots>
</ui>
